src/JWTAuthMiddleware.ts

```diff
@@ -147,6 +147,15 @@ export class JWTAuthMiddleware<Payload> {
   ): string | undefined {
     this.logger('Checking whether event contains authorization header')
     if (!isAuthorizedEvent(event)) {
+      if (this.options.authorizationRequired) {
+        throw createHttpError(
+          401,
+          'No valid bearer token was set in the authorization header',
+          {
+            type: 'AuthenticationRequired'
+          }
+        )
+      }
       this.logger('No authorization header found')
```

Comment:
The logging should happen before the error is thrown, otherwise it doesn't happen.

Reply:
It will happen in case credentialsRequired isn't set to true, we could also remove it altogether?

Reply:
No I mean that the line should be moved above the new code so it gets called in all cases :)

Reply:
I see, I thought that throwing the 401 would be enough

Reply:
fixed :)